sources/api/netdog/src/main.rs

```diff
@@ -280,6 +294,38 @@ fn generate_hostname() -> Result<()> {
     Ok(print_json(hostname)?)
 }
 
+/// Generate configuration for network interfaces.
+fn generate_net_config() -> Result<()> {
```

Comment:
I think this whole thing becomes test-able if you take the constants as input.

```suggestion
fn generate_net_config(net_config: P1, wicked_config_dir: P2, wicked_ext: S) -> Result<()> {
```

Reply:
I'm not sure it's necessary to have a unit test for this function.  Other than the filesystem operations (which I don't really see as needing to be tested), this function basically drives the other module functions.  The module functionality is tested in the `wicked` module!

Reply:
One could always make that argument! I'm seeing a fair amount of logic here "if the file exists do x", "if interfaces is set and dhcp4 is not set raise an error).